---
title: "Analysis Workflow"
author: "A. Cassano"
date: "2023-09-18"
output:   pdf_document:
    fig_width: 6.25
    keep_tex: yes
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Introduction
- This code should be useable to analyze RNA-seq data. Please read through the document before making any changes. This is a template and will indicate where you need to make changes for your specific experiment. 
- This R markdown file is broken down into "code chunks" that can be run individually. 
  - Downstream chunks rely on upstream chunks so running code out of order is not recommended. 
  - In R Studio, you can run a chunk by hitting the "play button" on the top right of the chunk. 
  - You can run all upstream chunks by clicking the button that has a down arrow pointing to a green bar.

Built with R version `r getRversion()`. Most recent template changes made on 

# Prepare work-space

## Install Packages 
- This segment of code should only be run if the packages are not already installed on whichever computer this is being run on.
- In order to have this chunk run, change `eval=FALSE` to `eval=TRUE` in the chunk header.
- If you're running chunk by chunk be sure to run this chunk the first time but do not run this code subsequent times.
- **Do not edit this chunk (other than changing eval to true)**
```{r install_pkgs, eval=FALSE, echo=FALSE}
install.packages(c("BiocManager",
                   "tidyverse",
                   "gplots",
                   "ggplot2",
                   "ggrepel",
                   "stringi",
                   "pheatmap",
                   "RColorBrewer"))

BiocManager::install(c("DESeq2",
                       "GO.db",
                       "GOstats",
                       "pathview",
                       "gage",
                       "gageData",
                       "GenomicRanges",
                       "Repitools",
                       "clusterProfiler",
                       "DOSE",
                       "pathview",
                       "treemap",
                       "ashr"))
BiocManager::install("clusterProfiler")
BiocManager::install("pathview")
BiocManager::install("enrichplot")
if (!require(devtools)) {
  install.packages('devtools')
}
devtools::install_github('JamesJoly/DGSEA')
devtools::install_github('amcassano/RNAseqAnalysisPackage')
```

## Set working directory 
- This chunk will set the 'current working directory'. You must change this to reflect what folder you'll be working in.
- The working directory is all where all outputs will be saved. All files being imported should also be in this folder.
- **Your working directory should contain:**
  - **CSV file containing all of the raw read counts obtained from the fastq files**
  - **CSV file containing the meta-data for your experiment**
  - **CSV files with any gene sets 
- Change the working directory variable (change what is in the quotes "") to reflect your current working directory. You will need to use the complete path.
- The path should end with the name of the folder, you should omit the trailing '/'
  - If you're using a Mac you can find the path to the folder ...
  - If you're using a PC you can find the path to the folder by ...  

```{r workingdirectory}
# set working directory
cwd <- "parentfolder/childfolder/yourfolder"
setwd(cwd)
```
